x-pack/qa/multi-cluster-search-security/legacy-with-basic-license/build.gradle

```diff
@@ -0,0 +1,76 @@
+import org.elasticsearch.gradle.internal.info.BuildParams
+import org.elasticsearch.gradle.internal.test.RestIntegTestTask
+
+apply plugin: 'elasticsearch.standalone-rest-test'
+apply plugin: 'elasticsearch.rest-resources'
+
+restResources {
+  restApi {
+    include '_common', 'bulk', 'field_caps', 'security', 'search', 'clear_scroll', 'scroll', 'async_search', 'cluster',
+      'indices', 'open_point_in_time', 'close_point_in_time', 'terms_enum'
+  }
+}
+
+// randomise between sniff and proxy modes
+boolean proxyMode = (new Random(Long.parseUnsignedLong(BuildParams.testSeed.tokenize(':').get(0), 16))).nextBoolean()
+
+def fulfillingCluster = testClusters.register('fulfilling-cluster') {
+  setting 'xpack.security.enabled', 'true'
+  setting 'xpack.license.self_generated.type', 'basic'
+  module ':modules:analysis-common'
+  module ':modules:lang-painless'
+  module ':modules:data-streams'
+  module ':x-pack:plugin:mapper-constant-keyword'
+  module ':x-pack:plugin:async-search'
+  user username: "test_user", password: "x-pack-test-password"
+}
+
+def queryingCluster = testClusters.register('querying-cluster') {
+  setting 'xpack.security.enabled', 'true'
+  setting 'xpack.license.self_generated.type', 'basic'
+  module ':modules:analysis-common'
+  module ':modules:lang-painless'
+  module ':modules:data-streams'
+  module ':x-pack:plugin:mapper-constant-keyword'
+  module ':x-pack:plugin:async-search'
+  setting 'cluster.remote.connections_per_cluster', "1"
+  user username: "test_user", password: "x-pack-test-password"
+
+  if (proxyMode) {
+    setting 'cluster.remote.my_remote_cluster.mode', 'proxy'
+    setting 'cluster.remote.my_remote_cluster.proxy_address', {
+      "\"${fulfillingCluster.get().getAllTransportPortURI().get(0)}\""
+    }
+  } else {
+    setting 'cluster.remote.my_remote_cluster.seeds', {
+      fulfillingCluster.get().getAllTransportPortURI().collect { "\"$it\"" }.toString()
+    }
+  }
+}
+
+tasks.register('fulfilling-cluster', RestIntegTestTask) {
+  useCluster fulfillingCluster
+  systemProperty 'tests.rest.suite', 'fulfilling_cluster'
+}
+
+tasks.register('querying-cluster', RestIntegTestTask) {
+  dependsOn 'fulfilling-cluster'
+  useCluster queryingCluster
+  useCluster fulfillingCluster
```

Comment:
To be clear this is a question for my edification, not a request for changes: What's the difference between `dependsOn 'fulfilling-cluster'` and `useCluster fulfillingCluster`? Like, obviously one is a testcluster and one is a task, but why do we need both `dependsOn` and `useCluster`?

Reply:
> Like, obviously one is a testcluster and one is a task, but why do we need both dependsOn and useCluster?

We actually don't need both for the first task, and for the second task we only need `useCluster fulfillingCluster` since the RestIntegTestTask (via elasticsearch.standalone-rest-test ->StandaloneRestTestPlugin -> RestTestBasePlugin) will automatically create/register a test cluster and [uses the task name](https://github.com/elastic/elasticsearch/blob/main/build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/test/RestTestBasePlugin.java#L58) with the same name.  I added here explicitly since, IMO it reads really weird and looks like you are using the wrong cluster without it defined.  

More generally you need `dependsOn` for the task graph since one task depends on task, and you need `useCluster` so that you have the explicit relationship to know when a test cluster can be shut down. It won't be shut down till all associated tasks have completed.